src/Factory.php

```diff
@@ -4,15 +4,20 @@
 
 use React\EventLoop\ExtUvLoop;
 use React\EventLoop\Loop;
-use React\EventLoop\LoopInterface;
 use React\Filesystem\Uv;
 use React\Filesystem\Eio;
-use React\Filesystem\ChildProcess;
 
 final class Factory
 {
     public static function create(): AdapterInterface
     {
+        /**
+         * ext-uv and ext-eio don't play well together so only using ext-eio adapter if we don't use ext-uv for the event loop
+         */
+        if (\function_exists('eio_get_event_stream') && !(Loop::get() instanceof ExtUvLoop)) {
+            return new Eio\Adapter();
+        }
```

Comment:
This is an interesting one! If both `ext-uv` and `ext-eio` are loaded, which filesystem extension should be used here and why?

Also, can you elaborate what "don't play well together" means exactly? Haven't tested this yet, but it's my understanding both libraries use a thread pool and should be independent to a certain degree, no?

Reply:
Need to do a very deep dive into why they won't play well together. But when running tests for both adapters in the same process. Tests blow up. Will get back on this

Reply:
It took me a while and discovered that my initial assumption of ext-eio and ex-uv not playing well together is wrong. During testing the ext-eio adapter is created twice. As such `eio_get_event_stream` is also called twice and that causes issues seemingly invalidating one of the streams we got from that function. After creating a singleton to get that stream all tests are now passing as expected.